### ISSUE TITLE
Activity page - General Stationary Combustion (non-compression and non-processing)

### ISSUE THREAD

**Issue by dleard** [bcgov/cas-reporting#303]
#### Description:

[Wireframe](https://www.figma.com/design/LsDmLDCdnJqI8UahyqvOD8/BCIERS-Designs?node-id=7837-20066&t=WU1VPKoUDpbR5KCv-4): 

Following #253 we have a full stack example of how to generate an activity emissions form using the API form builder service.The implementation here will be really similar to GSC - without line tracing. Most of the needed database records will already exist.

Documentation on the full stack steps necessary to create an activity page are in our [docs](https://github.com/bcgov/cas-registration/tree/develop/docs/frontend)

Use the excel file (BCIERS Reporting Fields V2 in the SME channel files  - Reporting App directory) provided by IRC to determine what the necessary activity, source type, gas type, methodology, reporting fields & related configurations that need to be added are.

#### Acceptance Criteria:

Given I am on the GSC non-compression activity form page
When I first land on the page
Then the activity schema is fetched & displayed

Given the schema has rendered correctly
When I make selections that have dependencies (gas type, methodology)
Then the correct options and fields as defined from the configuration are shown

#### Development Checklist:

- [ ] Any activities, source types, gas types, methodologies or reporting fields needed for the configuration that are not already in the database are added
- [ ] The configuration relationships have correctly been added to the database
- [ ] The many to many reporting field data (if needed) has correctly been added to the database
- [ ] The base schema for the activity & source types have been added to the database
- [ ] A routing directory/page has been added to both the industry_user & industry_user_admin flows for the activity
- [ ] A parent component (named for the activity) has been added to the components in the activity directory
- [ ] The page fetches the correct initial activity schema from the API service on initial load
- [ ] A uiSchema exists in the components/activities/uiSchema directory for the activity & source types (will move to the database eventually, but can live here for now)
- [ ] Conditional fields render dynamically as data is entered
- [ ] On selecting a source type (or if there is only one, on loading the page) The first set of data fields is displayed (defaultEmptySourceTypeState)
- [ ] Matches the wireframe

**Definition of Ready** (Note: If any of these points are not applicable, mark N/A)

- [ ] User story is included
- [ ] User role and type are identified
- [ ] Acceptance criteria are included
- [ ] Wireframes are included (if required)
- [ ] Design / Solution is accepted by Product Owner
- [ ] Dependencies are identified (technical, business, regulatory/policy)
- [ ] Story has been estimated (under 13 pts)

**Definition of Done** (Note: If any of these points are not applicable, mark N/A)

- [ ] Acceptance criteria are tested by the CI pipeline
- [ ] UI meets accessibility requirements
- [ ] Configuration changes are documented, documentation and designs are updated
- [ ] Passes code peer-review
- [ ] Passes QA of Acceptance Criteria with verification in Dev and Test
- [ ] Ticket is ready to be merged to main branch
- [ ] Can be demoed in Sprint Review
- [ ] Bugs or future work cards are identified and created
- [ ] Reviewed and approved by Product Owner

#### Notes:

-

**Comment by dleard** [comment]
temporarily blocking this as there are going to be 2 source types added to this activity. Should get an update on this soon & we can unblock it

**Comment by patriciarussellCAS** [comment]
@pbastia can you please share the URL so I can review? Thanks!

**Comment by pbastia** [comment]
@patriciarussellCAS yes! This should be deployed at /reporting/reports/1/facilities/1/activities/gsc_non_compression_non_processing

**Comment by patriciarussellCAS** [comment]
Tagging @suhafa for design review
My notes: 
- The title should match the spreadsheet, so instead of "General stationary non-compression and non-processing combustion" it should read: "General Stationary Combustion (non-compression and non-processing)"
- Need to confirm with BA whether "process vent gas combustion" is the appropriate wording for the 3rd option
- GCS Unit Type: This should be a drop-down
- Under Fuel 1:  We're missing Fuel Classification (drop-down menu with 4 options)
- Fuel Unit: Should be a drop-down

@pbastia

**Comment by patriciarussellCAS** [comment]
Review notes: @pbastia 
- still need to update the title to: General Stationary Combustion (non-compression and non-processing)
- Fuel unit and fuel classification are supposed to be dropdowns 
-  the second option appears to be broken (see screenshot)

![Image](https://github.com/user-attachments/assets/dd8a801b-b05c-4c43-9eb8-770bcafe634a)

**Comment by dleard** [comment]
The bug & the unit/classification fields have been fixed/handled in #349 (pending dev review). Only the fuel name will be a dropdown, classification and units will automatically be fetched from the database on fuel selection & shown in read-only fields

**Comment by suhafa** [comment]
**Design review:**
Not much I can add that hasn't already been captured, but I think we need to ensure the 'Add Emissions' button falls in line with the 'Emissions' information as the hierarchy of information and action is confusing currently in the app. Here is what it looks like in the designs:
![Image](https://github.com/user-attachments/assets/2c7085f3-8ae0-478d-9be1-0f510baf69fc)
Here is what it is currently looking like:
![Image](https://github.com/user-attachments/assets/1e9e6bec-158d-4efe-b83a-530f4e0d3f3d)

The distinction needs to be made clear

**Comment by suhafa** [comment]
Another note, upon selection of the second item, the page extends out far and wide. It all needs to be a fixed width:
![Image](https://github.com/user-attachments/assets/d3a67eb9-b6fd-4874-82b6-68f2c9600e63)

It is also compressing the task list awkwardly
![Image](https://github.com/user-attachments/assets/1cf7babb-3254-4fe8-99d7-8fa09f66fe47)

**Comment by patriciarussellCAS** [comment]
Still getting the error when selecting the 2nd checkbox. Will wait for #349 to get through before approving.
